src/apis/web3api.py

```diff
@@ -0,0 +1,215 @@
+"""
+Web3API for fetching relevant data using the web3 library.
+"""
+from os import getenv
+from typing import Dict, List, Tuple, Any, Optional
+from fractions import Fraction
+from dotenv import load_dotenv
+from web3 import Web3
+from web3.types import TxData, TxReceipt
+from eth_typing import Address, HexStr
+from hexbytes import HexBytes
+from contracts.gpv2_settlement import gpv2_settlement as gpv2Abi
+from src.trades import Trade, OrderData, OrderExecution
+from src.helper_functions import get_logger
+from src.constants import (
+    ADDRESS,
+)
+
+
+class Web3API:
+    """
+    Class for fetching data from a Web3 API.
+    """
+
+    def __init__(self):
+        load_dotenv()
+        infura_key = getenv("INFURA_KEY")
+        self.url = f"https://mainnet.infura.io/v3/{infura_key}"
+        self.web_3 = Web3(Web3.HTTPProvider(self.url))
+        self.contract = self.web_3.eth.contract(
+            address=Address(HexBytes(ADDRESS)), abi=gpv2Abi
```

Comment:
ubernit - this `gpv2Abi` follows an incredibly awkward naming convention that feels like it wants to be `camelCase` but isn't really even though python uses `snake_case` and seems like it should be a `CONSTANT`

Reply:
(I did something random here. Since the changes might have to be backwards compatible, I held off changing this name.)

Reply:
oh I see. no worries. It was just an UBERNIT